mat64/eigen.go

```diff
@@ -23,12 +23,61 @@ func symmetric(m *Dense) bool {
 	return true
 }
 
-type EigenFactors struct {
+// Eigen is a type for creating an using the eigenvalue decomposition of a amtrix.
+type Eigen struct {
+	vectorsComputed bool
+
+	n int
+
+	ef eigenFactors
+}
+
+// Factorize computes the eigenvalue decompositon of the input square matrix a.
+// The eigenvalue decomposition is defined as
+//  A = P * D * P^-1
+// where D is a diagonal matrix containing the eigenvalues of the matrix, and
+// P is a matrix of the eigenvectors of A. If the vectors input argument is
+// false, the eigenvectors are not computed.
```

Comment:
This is not true at the moment. `vectors` is assigned to `vectorsComputed` which is then never used.